Core/Src/state_machine.c

```diff
@@ -100,6 +104,8 @@ static int transition_functional_state(func_state_t new_state, pdu_t *pdu,
 		/* Turn off high power peripherals */
 		// write_fan_battbox(pdu, true);
 		write_pump(pdu, false);
+		// delay for one second before faulting the car 
+		osDelay(pdMS_TO_TICKS(1000));
```

Comment:
We should transition MPU to faulted state before this delay so that the controls are in faulted state. We should do this because we are only interested in protecting the faulting hardware, so we can (and should) change MPU internal state with no delay. We want to change MPU state asap because we don't want to be issuing commands to the motor controller if MPU is faulted (FAULTED state stops commands to the motor controller).